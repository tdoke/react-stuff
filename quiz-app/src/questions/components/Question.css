.question {
  width: 400px;
}
.question .question-label {
  text-align: justify;
  line-height: 32px;
}