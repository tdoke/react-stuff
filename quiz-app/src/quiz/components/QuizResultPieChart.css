.pie-chart-container {
  box-sizing: border-box;
}
.pie-chart-container ul.pie-chart-colors {
  list-style: none;
}
.pie-chart-container .chart-stat {
  font-size: 0.9em;
}
div.color-box {
  display: inline-block;
  width: 15px;
  height: 15px;
}
div.color-box.correct {
  background-color: #00C49F;
}
div.color-box.wrong {
  background-color: #FF0000;
}
div.color-box.not-attempted {
  background-color: #FFBB28;
}