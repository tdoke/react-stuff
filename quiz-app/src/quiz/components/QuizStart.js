import React from 'react';
import Button from '@material-ui/core/Button';
import FlexView from 'react-flexview/lib';

export const QuizStart = ({ onStartBtnClick }) => (
  <div>
    <FlexView column hAlignContent='center' marginTop={48}>
      <FlexView>
        <h1>Cricket Quiz</h1>
      </FlexView>
      <FlexView column marginTop={32}>
        <FlexView marginBottom={16}>
          <FlexView hAlignContent='left' marginRight={32}>Total Questions</FlexView>
          <FlexView hAlignContent='right' marginLeft={32}>10</FlexView>
        </FlexView>
        <FlexView>
          <FlexView hAlignContent='left' marginRight={32} grow>Passing</FlexView>
          <FlexView hAlignContent='right' marginLeft={32} basis={16}>80%</FlexView>
        </FlexView>
      </FlexView>
      <FlexView column hAlignContent='center' marginTop={48}>
        <Button
          variant="contained"
          color="primary"
          aria-label="start"
          onClick={onStartBtnClick}>
          START
            </Button>
      </FlexView>
    </FlexView>
  </div>
)
