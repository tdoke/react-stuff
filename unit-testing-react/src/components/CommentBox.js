import React, { Component } from 'react';

class CommentBox extends Component {
  constructor(props) {
    super(props);
    this.state = {
      comments: ''
    }
  }

  handleInputChange = (event) => {
    this.setState({ comments: event.target.value })
  }

  handleSubmitBtnClick = (event) => {
    event.preventDefault()
    // call an action saving comments

    this.setState({ comments: '' })
  }

  render() {
    return (<div>
      <h4>add comment</h4>
      <form onSubmit={this.handleSubmitBtnClick}>
        <textarea value={this.state.comments} onChange={this.handleInputChange}></textarea>
        <button>submit</button>
      </form>
    </div>);
  }
}

export default CommentBox;