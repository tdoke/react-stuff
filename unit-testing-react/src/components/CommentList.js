import React from 'react';

const commenList = () => {
  return ( <div>i am commen list</div> );
}
 
export default commenList;