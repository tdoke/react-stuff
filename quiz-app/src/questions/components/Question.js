import React from 'react';
import FlexView from 'react-flexview/lib';
import Radio from '@material-ui/core/Radio';
import RadioGroup from '@material-ui/core/RadioGroup';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import FormControl from '@material-ui/core/FormControl';
import Button from '@material-ui/core/Button';
import './Question.css';

export const Question = ({ 
  id,
  questionText,
  options,
  usersAnswer,
  isLast,
  onAnswerRadioBtnClick,
  onLastAnswerSubmit }) => (

    <div className="question">
      <FlexView marginBottom={16} marginTop={48} hAlignContent="left">
        <label className="question-label">{questionText}</label>
      </FlexView>
      <FlexView marginBottom={16} hAlignContent="left">
        <FormControl component="fieldset">
          <RadioGroup
            aria-label="question"
            name="question"
            value={String(usersAnswer)}
            onChange={(event) => onAnswerRadioBtnClick(id, Number(event.target.value))}>
            {
              options.map(option =>
                <FormControlLabel
                  key={option.id}
                  value={String(option.id)}
                  control={<Radio />}
                  label={option.text} />
              )
            }
          </RadioGroup>
        </FormControl>
      </FlexView>
      { isLast && 
      <FlexView hAlignContent="center">
        <Button
          variant="contained"
          color="primary"
          aria-label="start"
          onClick={onLastAnswerSubmit}>
          SUBMIT
      </Button>
      </FlexView>
      }
    </div>
  )