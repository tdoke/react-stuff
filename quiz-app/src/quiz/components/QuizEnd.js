import React from 'react';
import FlexView from 'react-flexview/lib';
import QuizResultPieChart from './QuizResultPieChart';

const calculateScore = (questions) => {
  const totalQuestions = questions.length;
  const correctQuestions = questions.filter(q => q.answer === q.usersAnswer).length;
  return 100 * correctQuestions / totalQuestions;
}

export const QuizEnd = ({ questions }) => (
  <div>
    <FlexView column hAlignContent='center' marginTop={48}>
      <FlexView>
        <h1>Cricket Quiz Result</h1>
      </FlexView>
      <FlexView column marginTop={16}>
        <FlexView marginBottom={16} hAlignContent='center'>
          <FlexView hAlignContent='left' marginRight={32} style={{ width: '40%' }}>Total Questions</FlexView>
          <FlexView hAlignContent='right' marginLeft={32} style={{ width: '40%' }}>{questions.length}</FlexView>
        </FlexView>
        <FlexView marginBottom={16} hAlignContent='center'>
          <FlexView hAlignContent='left' marginRight={32} style={{ width: '40%' }}>Score</FlexView>
          <FlexView hAlignContent='right' marginLeft={32} style={{ width: '40%' }}>{calculateScore(questions)}&nbsp;%</FlexView>
        </FlexView>
        <FlexView>
          <QuizResultPieChart questions={questions} />
        </FlexView>
      </FlexView>
    </FlexView>

  </div >
)
