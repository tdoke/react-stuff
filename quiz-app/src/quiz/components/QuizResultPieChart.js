import React from 'react'
import { ResponsiveContainer, PieChart, Pie, Cell } from 'recharts';
import FlexView from 'react-flexview/lib';
import './QuizResultPieChart.css';

const COLORS = {
  correct: '#00C49F',
  wrong: '#FF0000',
  skipped: '#FFBB28'
}

const formatToPieChartInput = (questions) => ([
  {
    name: 'Correct',
    value: questions.filter(q => q.answer === q.usersAnswer).length
  },
  {
    name: 'Wrong',
    value: questions.filter(q => q.usersAnswer !== 0 && q.answer !== q.usersAnswer).length
  },
  {
    name: 'Skipped',
    value: questions.filter(q => q.usersAnswer === 0).length
  }
])

const formatToPercentage = (questions) => (
  {
    correct: 100 * questions.filter(q => q.answer === q.usersAnswer).length / questions.length,
    wrong: 100 * questions.filter(q => q.usersAnswer !== 0 && q.answer !== q.usersAnswer).length / questions.length,
    skipped: 100 * questions.filter(q => q.usersAnswer === 0).length / questions.length
  }
)
const QuizResultPieChart = ({ questions }) => {
  const data = formatToPieChartInput(questions)
  const { correct, wrong, skipped } = formatToPercentage(questions)
  return (
    <div className='pie-chart-container'>
      <FlexView>
        <FlexView hAlignContent='left'>
          <PieChart height={200} width={300}>
            <Pie
              data={data}
              dataKey="value"
              cx={90}
              cy={90}
              innerRadius={60}
              outerRadius={80}
              fill="#8884d8"
              paddingAngle={5}>
              {
                data.map((entry, index) => <Cell key={entry.name} fill={COLORS[entry.name.toLowerCase()]} />)
              }
            </Pie>
          </PieChart>
        </FlexView>
        <FlexView column hAlignContent='left' marginTop={40} className='chart-stat'>
          <FlexView marginBottom={8} style={{ width: '100%' }}>
            <FlexView hAlignContent='left' vAlignContent='center' marginRight={8} style={{ width: '20%' }}><div className='color-box correct'></div></FlexView>
            <FlexView hAlignContent='left' vAlignContent='center' style={{ width: '60%' }} marginRight={16}>Correct</FlexView>
            <FlexView hAlignContent='right' vAlignContent='center' style={{ width: '20%' }}>{correct}%</FlexView>
          </FlexView>
          <FlexView marginBottom={8} style={{ width: '100%' }}>
            <FlexView hAlignContent='left' vAlignContent='center' marginRight={8} style={{ width: '20%' }}><div className='color-box wrong'></div></FlexView>
            <FlexView hAlignContent='left' vAlignContent='center' style={{ width: '60%' }} marginRight={16}>Wrong</FlexView>
            <FlexView hAlignContent='right' vAlignContent='center' style={{ width: '20%' }}>{wrong}%</FlexView>
          </FlexView>
          <FlexView marginBottom={8} style={{ width: '100%' }}>
            <FlexView hAlignContent='left' vAlignContent='center' marginRight={8} style={{ width: '20%' }}><div className='color-box not-attempted'></div></FlexView>
            <FlexView hAlignContent='left' vAlignContent='bottom' style={{ width: '60%' }} marginRight={16}>Not Attempted</FlexView>
            <FlexView hAlignContent='right' vAlignContent='center' style={{ width: '20%' }}>{skipped}%</FlexView>
          </FlexView>
        </FlexView>
      </FlexView>
    </div>
  )
}

export default QuizResultPieChart;
