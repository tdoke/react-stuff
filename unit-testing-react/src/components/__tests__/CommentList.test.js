import React from 'react'
import { shallow } from 'enzyme'
import { CommentsList } from 'components/CommentList'

let wrapper
let props

beforeEach(() => {
  props = {
    comments: [] // provide default props here or else wrapper becomes undefined
  }
  wrapper = shallow(<CommentsList {...props} />)
})

afterEach(() => {
  wrapper.unmount()
})

it('should show comments in a list', () => {
  const commentsData = ['test11', 'test2']
  wrapper.setProps({
    comments: commentsData
  })
  expect(wrapper.find('li').length).toEqual(2)
  expect(wrapper).toMatchSnapshot()
})
