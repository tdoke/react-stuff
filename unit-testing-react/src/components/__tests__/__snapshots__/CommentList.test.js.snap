// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show comments in a list 1`] = `ShallowWrapper {}`;
